engine/src/main/java/com/datatorrent/stram/StreamingContainerManager.java

```diff
@@ -780,17 +780,17 @@ public void monitorHeartbeat(boolean waitForRecovery)
           //LOG.debug("{} {} {}", c.getExternalId(), currentTms - sca.createdMillis, this.vars.heartbeatTimeoutMillis);
           // container allocated but process was either not launched or is not able to phone home
           if (currentTms - sca.createdMillis > 2 * this.vars.heartbeatTimeoutMillis) {
-            LOG.error("Container {}@{} startup timeout ({} ms).", c.getExternalId(), c.host, currentTms - sca.createdMillis);
+            LOG.warn("Container {}@{} startup timeout ({} ms).", c.getExternalId(), c.host, currentTms - sca.createdMillis);
             containerStopRequests.put(c.getExternalId(), c.getExternalId());
           }
         } else {
           if (currentTms - sca.lastHeartbeatMillis > this.vars.heartbeatTimeoutMillis) {
             if (!isApplicationIdle()) {
               // Check if the heartbeat for this agent has already been missed to raise the StramEvent only once
               if (sca.lastHeartbeatMillis != -1) {
-                String info = String.format("Container %s@%s heartbeat timeout (%d%n ms).", c.getExternalId(), c.host, currentTms - sca.lastHeartbeatMillis);
-                LOG.error(info);
-                StramEvent stramEvent = new StramEvent.ContainerErrorEvent(c.getExternalId(), info, null);
+                String info = String.format("Container %s@%s heartbeat timeout  (%d%n ms).", c.getExternalId(), c.host, currentTms - sca.lastHeartbeatMillis);
+                LOG.warn(info);
+                StramEvent stramEvent = new StramEvent.ContainerErrorEvent(c.getExternalId(), info, null, StramEvent.LogLevel.WARN);
```

Comment:
logs are created not for the apex engine, but for dev-ops and apex developers to analyze and troubleshoot problems both in the engine itself and in the run-time environments. It will be confusing for users to get ContainerErrorEvent and not to have any indication of the problem in the AppMaster logs.

 To me, this is similar to a checked exception in Java. Checked exceptions are handled by an application, but they are still exceptions and need to be logged as errors or warnings.

I was not referring to the StramEvent level, the event that is raise is Container*Error*Event and it should use default level for the Event.